src/study.js

```diff
@@ -142,86 +119,38 @@ class Study {
           }
           weights.push(weight);
         });
-        bucket = chooseWeightedItem(names, weights);
-      }
 
-      if (shouldRemoveBucket) {
-        this.classList[test.name] = null;
-        delete this.userAssignments[test.name];
-      } else {
-        // Add to our assignments
-        this.classList[test.name] = bucket;
-        this.userAssignments[test.name] = bucket;
+        persistedUserAssignments[test.name] =
+        userAssignments[test.name] = chooseWeightedItem(names, weights);
       }
     });
 
-    if (!isServer && document.body) applyClasses();
-    else document.addEventListener('DOMContentLoaded', applyClasses);
+    this.persist();
+    this.applyClasses();
+  }
 
-    // Persist buckets
-    this.persist(this.userAssignments);
+  assign(testName, bucketName) {
+    if (!testName) return this.assignAll();
+
+    const test = this.providedTests.filter(x => x.name === testName)[0];
+    if (bucketName === null || !test) {
+      delete this.userAssignments[testName];
+      delete this.persistedUserAssignments[testName];
+      this.removeClasses(testName);
+    } else {
+      this.userAssignments[testName] =
```

Comment:
i would refactor this to be more readable. took a lot of my mental energy to grok what was going on here.